lib/_stream_readable.js

```diff
@@ -467,6 +467,9 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
               dest !== process.stdout &&
               dest !== process.stderr;
 
+  var autoCork = pipeOpts && pipeOpts.cork && (typeof dest.cork === 'function');
```

Comment:
Is there a time when `dest.cork` isn't a function? Won't it error anyways if it isn't a writable stream?